src/ImageProcessorCore/Formats/Tiff/TiffReader.cs

```diff
@@ -0,0 +1,232 @@
+﻿using System;
+using System.IO;
+using System.Text;
+
+namespace ImageProcessorCore.Formats.Tiff
+{
+    /// <summary>
+    /// Encapsulates a seekable stream of bytes.
+    /// </summary>
+    /// <remarks>
+    /// This class has several helper or conversion methods that take raw bytes and converts them
+    /// to well known .net types. The <see cref="BitOrder"/> can be changed to support Little Endian (intel) 
+    /// or Big Endian (motorolla)
+    /// </remarks>
+    public class TiffReader
```

Comment:
I have some classes that I'd really like to use throughout for handling endianess. (I'm not yet though my jpeg encoder in the jpeg branch uses the EndianBinaryWriter) 

https://github.com/JimBobSquarePants/ImageProcessor/tree/17cd44fdcf56bd26a2a22ddecc156c9c35ae6478/src/ImageProcessorCore/IO

This should help reduce the amount of code you have to write.